runtime/Cpp/runtime/src/Parser.cpp

```diff
@@ -214,25 +229,29 @@ TokenFactory<CommonToken>* Parser::getTokenFactory() {
 
 
 const atn::ATN& Parser::getATNWithBypassAlts() {
-  std::vector<uint16_t> serializedAtn = getSerializedATN();
+  const std::vector<uint16_t> &serializedAtn = getSerializedATN();
   if (serializedAtn.empty()) {
     throw UnsupportedOperationException("The current parser does not support an ATN with bypass alternatives.");
   }
 
-  std::lock_guard<std::mutex> lck(_mutex);
-
-  // XXX: using the entire serialized ATN as key into the map is a big resource waste.
-  //      How large can that thing become?
-  if (bypassAltsAtnCache.find(serializedAtn) == bypassAltsAtnCache.end())
+  auto *cache = getBypassAltsAtnCache();
   {
```

Comment:
You removed my comment, which I think is still valid. A potentially large string is used as key for the bypass cache. With that comment I wanted to express that we should look for a less resource intensive solution.

Reply:
Added back.